storage/remote/otlptranslator/prometheusremotewrite/helper.go

```diff
@@ -521,51 +449,50 @@ func (c *PrometheusConverter) addSummaryDataPoints(ctx context.Context, dataPoin
 // createLabels returns a copy of baseLabels, adding to it the pair model.MetricNameLabel=name.
 // If extras are provided, corresponding label pairs are also added to the returned slice.
 // If extras is uneven length, the last (unpaired) extra will be ignored.
-func createLabels(name string, baseLabels []prompb.Label, extras ...string) []prompb.Label {
-	extraLabelCount := len(extras) / 2
-	labels := make([]prompb.Label, len(baseLabels), len(baseLabels)+extraLabelCount+1) // +1 for name
-	copy(labels, baseLabels)
+func (c *PrometheusConverter) createLabels(name string, baseLabels labels.Labels, extras ...string) labels.Labels {
+	c.builder.Reset(baseLabels)
 
 	n := len(extras)
 	n -= n % 2
 	for extrasIdx := 0; extrasIdx < n; extrasIdx += 2 {
-		labels = append(labels, prompb.Label{Name: extras[extrasIdx], Value: extras[extrasIdx+1]})
+		c.builder.Set(extras[extrasIdx], extras[extrasIdx+1])
 	}
-
-	labels = append(labels, prompb.Label{Name: model.MetricNameLabel, Value: name})
-	return labels
+	c.builder.Set(model.MetricNameLabel, name)
+	return c.builder.Labels()
 }
 
 // getOrCreateTimeSeries returns the time series corresponding to the label set if existent, and false.
 // Otherwise it creates a new one and returns that, and true.
-func (c *PrometheusConverter) getOrCreateTimeSeries(lbls []prompb.Label) (*prompb.TimeSeries, bool) {
-	h := timeSeriesSignature(lbls)
+func (c *PrometheusConverter) getOrCreateTimeSeries(lbls labels.Labels, metadata writev2.Metadata) (*writev2.TimeSeries, bool) {
+	h := lbls.Hash()
 	ts := c.unique[h]
 	if ts != nil {
-		if isSameMetric(ts, lbls) {
+		if labels.Equal(ts.ToLabels(&c.scratchBuilder, c.symbolTable.Symbols()), lbls) {
 			// We already have this metric
 			return ts, false
 		}
 
 		// Look for a matching conflict
```

Comment:
```suggestion
		// Look for a matching conflict.
```
.. and in the whole of this function 🙈